docs/en/install-upgrade/installing-stack-demo-secure.asciidoc

```diff
@@ -0,0 +1,894 @@
++
+["source","shell"]
+----
+xpack.security.transport.ssl:
+   ...
+   keystore.path: /etc/elasticsearch/certs/elastic-certificates.p12
+   truststore.path: /etc/elasticsearch/certs/elastic-stack-ca.p12
+----
+
+[discrete]
+[[install-stack-demo-secure-transport-es-keystore]]
+=== Update the {es} keystore
+
+{es} uses a separate keystore to hold the passwords of the keystores and truststores holding the CA and node certificates created in the previous steps. Access to this keystore is through the use of a utility called `elasticsearch-keystore`.
+
+. From the {es} installation directory, list the contents of the existing keystore:
++
+["source","shell"]
+----
+/usr/share/elasticsearch/bin/elasticsearch-keystore list
+----
++
+The results should be like the following:
++
+["source","yaml"]
+----
+keystore.seed
+xpack.security.http.ssl.keystore.secure_password
+xpack.security.transport.ssl.keystore.secure_password
+xpack.security.transport.ssl.truststore.secure_password
+----
++
+Notice that there are entries for:
++
+* The `transport.ssl.truststore` that holds the CA certificate
+* The `transport.ssl.keystore` that holds the CA-signed certificates
+* The `http.ssl.keystore` for the HTTP layer
++
+These entries were created at installation and need to be replaced with the passwords to the newly-created truststore and keystores.
+
+. Remove the existing keystore values for the default transport keystore and truststore:
++
+["source","shell"]
+----
+sudo usr/share/elasticsearch/bin/elasticsearch-keystore remove xpack.security.transport.ssl.keystore.secure_password
+
+sudo usr/share/elasticsearch/bin/elasticsearch-keystore remove xpack.security.transport.ssl.truststore.secure_password
+----
+
+. Update the `elasticsearch-keystore` with the passwords for the new keystore and truststore created in the previous steps. This ensures that {es} can read the new stores:
++
+["source","shell"]
+----
+sudo usr/share/elasticsearch/bin/elasticsearch-keystore add xpack.security.transport.ssl.keystore.secure_password
+
+sudo usr/share/elasticsearch/bin/elasticsearch-keystore add xpack.security.transport.ssl.truststore.secure_password
+----
+
+[discrete]
+[[install-stack-demo-secure-http]]
+== Step 3: Generate new certificate(s) for the HTTP layer
+
+Now that communication between {es} nodes (the transport layer) has been secured with SSL certificates, the same must be done for the communications that use the REST API, including {kib}, clients, and any other components on the HTTP layer.
+
+. Similar to the process for the transport layer, on the first node in your {es} cluster use the certificate utility to generate a CA certificate for HTTP communications:
++
+["source","shell"]
+----
+sudo / usr/share/elasticsearch/bin/elasticsearch-certutil http
+----
++
+Respond to the command prompts as follows:
+
+* When asked if you want to generate a CSR, enter `n`.
+* When asked if you want to use an existing CA, enter `y`.
```

Comment:
Maybe put a note here that if using their organization's CA certificate the following two steps would refer to that certificate and key.